src/Matcher/RecordedEventsThatMatcher.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare (strict_types = 1);
+
+namespace GQuemener\PhpSpec\Matcher;
+
+use PhpSpec\Matcher\BasicMatcher;
+use PhpSpec\Formatter\Presenter\Presenter;
+use PhpSpec\Exception\Example\FailureException;
+
+final class RecordedEventsThatMatcher extends BasicMatcher
+{
+    /**
+     * @var Presenter
+     */
+    private $presenter;
+
+    /**
+     * @param Presenter $presenter
+     */
+    public function __construct(Presenter $presenter)
+    {
+        $this->presenter = $presenter;
+    }
+
+    /**
+     * @param string $name
+     * @param mixed  $subject
+     * @param array  $arguments
+     *
+     * @return bool
+     */
+    public function supports(string $name, $subject, array $arguments): bool
+    {
+        return 'haveRecordedEventsThat' === $name
```

Comment:
Subject should be an `Prooph\EventSourcing\Aggregate\AggregateRoot` or use trait `Prooph\EventSourcing\Aggregate\EventProducerTrait`. I think the first would be enough to cover 99% of the cases.